integration/src/shopify-token-exchange/shopify-token-exchange.module.ts

```diff
@@ -0,0 +1,32 @@
+import { ShopifyAuthModule } from '@nestjs-shopify/auth';
+import { Module } from '@nestjs/common';
+import { MyAfterAuthHandler } from './my-after-auth-handler';
+import { MyAuthModule } from './my-auth.module';
+
+@Module({
+  imports: [
+    ShopifyAuthModule.forRootAsyncOnline(
+      {
+        imports: [MyAuthModule],
+        useFactory: (myAuthHandler: MyAfterAuthHandler) => ({
+          returnHeaders: true,
+          afterAuthHandler: myAuthHandler,
+        }),
+        inject: [MyAfterAuthHandler],
+      },
+      'TOKEN_EXCHANGE',
```

Comment:
with your latest changes, removing this line will NOT result in a TS error, although it should since the handler types are different for token exchange and auth code flow.

since we decided this to be a major change, we could also make the auth strategy a required parameter so we get typesafety:

e.g.
```ts
  static forRootAsyncOnline(
    options: ShopifyAuthModuleAsyncOptions<
      'AUTHORIZATION_CODE_FLOW',
      ShopifyAuthorizationCodeAuthModuleOptions
    >,
    authStrategy: 'AUTHORIZATION_CODE_FLOW',
  ): DynamicModule;
  static forRootAsyncOnline(
    options: ShopifyAuthModuleAsyncOptions<
      'TOKEN_EXCHANGE',
      ShopifyTokenExchangeAuthModuleOptions
    >,
    authStrategy: 'TOKEN_EXCHANGE',
  ): DynamicModule;
  static forRootAsyncOnline(
    options: ShopifyAuthModuleAsyncOptions<
      AuthStrategy,
      ShopifyAuthModuleOptions
    >,
    authStrategy: AuthStrategy,
  ): DynamicModule {
```

Reply:
@schiller-manuel it seems that `tsc` can still infer the correct method overload just by the options that we pass. That would be unwanted as we want someone to explicitly opt into one of these types. If you can think of a solution, feel free to take a look. Otherwise I will try something later 👍 .

Reply:
well the "go-to" solution typically is passing in a single object that is used as a discriminated union.
I had this initially by putting the `authStrategy` property into the options object.

This allowed TS to correctly infer everything based on this property.

now that we have two separate parameters, it becomes more complicated.
As I wrote above, making the parameter required solves the issue, but of course it makes this really breaking as now a user HAS TO specify an auth strategy.

Reply:
@schiller-manuel I just fixed it with backwards compatible overrides in `ShopifyAuthModule`. The types got quite exhausting though. Hopefully Shopify soon drops multiple flows so we can greatly simplify all this.

Thanks again for catching this 🙏 .